E2E: Limit themes in Gutenberg upgrade testing

WunderBart opened the pull request:
#### Changes proposed in this Pull Request

Run the GB upgrade test in only one theme/test site in order to reduce the running time. 

We were running the upgrade test for each block in 5 test sites (each with a different popular theme), which was making the test extremely lengthy, even when running in parallel. Editor glitches caused by themes are rare so we decided to get rid of the multi-theme aspect in the test, by reducing the test site/theme to one.

We'll (probably) resume testing on the most popular themes when we get to work on automated visual regression testing. It'll most probably be triggered by a different label on Github.

#### Testing instructions

##### Spec

* All checks/tests should pass for this PR
* (Optional) Run the spec locally: `cd wp-calypso/test/e2e && GUTENBERG_EDGE=true ./node_modules/.bin/mocha specs/wp-calypso-gutenberg-upgrade-spec.js`

##### Encrypted settings diff

The `themedSites` setting has been removed from the encrypted E2E config file. To sanity-check the diff, you can do the following:

1. `cd wp-calypso/test/e2e`
2.  Make sure your `local-decrypted.json` in `config/` is up to date;
3. `mv config/local-decrypted.json config/local-decrypted.prev.json`;
4. Decrypt the `encrypted.enc` file (refer to the fieldguide). It will create a new `local-decrypted.json`;
5. `diff config/local-decrypted.prev.json config/local-decrypted.json` and make sure only the `themedSites` key alongside its values have been removed;
6. You can also run `python -m json.tool config/local-decrypted.json` to make sure it's valid JSON (it will err with `No JSON object could be decoded` or a syntax error message in case of invalid JSON, or print a beautified JSON otherwise).

WunderBart committed:
Remove disabled code

WunderBart committed:
Move global vars to instance vars

WunderBart committed:
Do not test upgrade agains multiple themed sites

WunderBart committed:
Revert the driver global var

WunderBart committed:
Fix more vars

fullofcaffeine commented on test/e2e/specs/wp-calypso-gutenberg-upgrade-spec.js:
```diff
@@ -102,84 +96,11 @@ async function insertBlock( blockClass ) {
 	assert.strictEqual( errorDisplayed, false, 'The block errored in the editor!' );
 }
 
-// Commented-out for now because of https://github.com/Automattic/jetpack/issues/16514
```
I think we should reactivate the invalid block checks. The related issue is still open but it might worth double-checking to see if it's still failing.

fullofcaffeine replied:
This has been done ✅

fullofcaffeine commented on test/e2e/specs/wp-calypso-gutenberg-upgrade-spec.js:
```diff
@@ -198,16 +119,6 @@ function verifyBlockInEditor( blockClass ) {
-	// 		await assertNoInvalidBlocksInEditor();
-	// } );
-
-	// Commented out for now because of ENOMEM erros on CI
```
Good house-keeping 👍🏻  I don't think those screenshots were ever useful. Plus, we already have hooks to take a screenshot/video when tests fail. I think these would be ideally replaced by automated visual regression tests.

fullofcaffeine committed:
Remove the "themedSites" setting from the e2e config as it is not used anymore

fullofcaffeine committed:
Readd the check for invalid blocks in the editor

fullofcaffeine commented on test/e2e/specs/wp-calypso-gutenberg-upgrade-spec.js:
```diff
@@ -303,44 +201,46 @@ describe( `[${ host }] Test Gutenberg upgrade from non-edge to edge across most
 						await loginFlow.login( siteURL, true );
-						await startNewPost( siteURL );
-					} );
+						isLoggedIn = true;
```
👍🏻

WunderBart committed:
Add @parallel tag to actually run the tests :)

WunderBart commented on test/e2e/specs/wp-calypso-gutenberg-upgrade-spec.js:
```diff
@@ -102,83 +96,14 @@ async function insertBlock( blockClass ) {
-// 	);
-// }
-
 async function assertNoErrorInEditor() {
```
I'm wondering if it makes sense to keep this one. We already check if the block is present and if it doesn't invalidate. Does this actually cover another scenario or are we safe with presence and validation checks? What do you think, @fullofcaffeine? I'm asking because this one needs a full timeout to complete so it's adding quite a bit of time.

fullofcaffeine replied:
Hmm, good question. I'm not entirely sure, but it's worth checking. This actually checks if the editor crashed, not the block. However, if the editor crashes, the test will fail anyway, so I guess we could indeed remove this assertion. I'd leave it for another PR, though.

WunderBart committed:
Move parallel tag to individual block test scope

WunderBart committed:
Clean up; Add comments

fullofcaffeine reviewed (approved):
Looks good. If you could just sanity-check the changes I made to encrypted.enc I'd be grateful, @WunderBart :) thanks!

WunderBart committed:
Update e2e config

WunderBart committed:
Fix failing LayoutGrid block test

WunderBart committed:
Make test desc shorter (more readable)

WunderBart committed:
Reduce sample images cound to 3
Every image adds more time for the block to be rendered, which causes it to sometimes timeout actually. After reducing it to 3 it seems to be passing consistently for me.

WunderBart committed:
tmp: check if the suite fails at login for just one block (no parallel)

WunderBart committed:
tmp: check if the suite fails at login for 4 blocks in parallel

WunderBart committed:
tmp: check if the suite fails at login for 6 blocks in parallel

WunderBart committed:
tmp: check if the suite fails at login for 5 blocks in parallel

WunderBart committed:
tmp: double-check with 4 parallels

WunderBart committed:
tmp: run the suite in sync mode

WunderBart committed:
Revert tests

tyxla reviewed (approved):
LGTM 👍
